src/inspect_ai/util/_sandbox/environment.py

```diff
@@ -6,14 +8,23 @@
 
 from .._subprocess import ExecResult
 
-TaskInit = Callable[[str, str | None], Awaitable[None]]
-TaskCleanup = Callable[[str, str | None, bool], Awaitable[None]]
+TaskInit = Callable[[str, Union["SandboxEnvironmentConfigType", None]], Awaitable[None]]
```

Comment:
Sadly even `from __future__ import annotations` doesn't allow me to to forward reference `SandboxEnvironmentConfigType` here.

And `"SandboxEnvironmentConfigType" | None` is not supported, so I've explicitly used `Union`.